core/src/main/resources/filodb-defaults.conf

```diff
@@ -38,6 +38,9 @@ filodb {
 
     # Minimum step required for a query
     min-step = 5 seconds
+
+    # Number of threads in query threadpool.  There is one threadpool per dataset.
+    num-threads = 8
```

Comment:
I dont understand this. Each dataset has a queryActor. Why thread per dataset?
Why are we hard-coding this value ?

Reply:
Just to make sure we are on the same page - so there is now an explicit threadpool for all QueryEngine work for each dataset.  This parameter controls that threadpool size.  The QueryActor is just handling the API and passing off the work - it is doing next to nothing now. 
The whole point of this change is to offload work from the QueryActor receiving thread to a separate threadpool that we can easily profile and control.